packages/dashboard-frontend/src/store/FactoryResolver/devfile-service.ts

```diff
@@ -0,0 +1,203 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+export const cheDevfileServicePath = '/services/che-devfile-service';
+
+export const DevfileService = Symbol('DevfileService');
+
+export interface DevfileComponentStatus {
+  name: string;
+  isUser: boolean;
+  env?: DevfileComponentEnv[];
+  endpoints?: {
+    [endpointName: string]: {
+      url?: string;
+    };
+  };
+}
+export interface DevfileMetadata {
+  attributes?: { [attributeName: string]: string };
+  description?: string;
+  displayName?: string;
+  globalMemoryLimit?: string;
+  icon?: string;
+  name?: string;
+  tags?: string[];
+  version?: string;
+}
+
+export interface Devfile {
+  schemaVersion: string;
+  attributes?: { [attributeName: string]: string };
+  metadata: DevfileMetadata;
+  projects?: DevfileProject[];
+  components?: DevfileComponent[];
+  commands?: DevfileCommand[];
+}
+export interface DevfileCommandGroup {
+  isDefault?: boolean;
+  kind: 'build' | 'run' | 'test' | 'debug';
+}
+
+export interface DevfileCommand {
+  apply?: {
+    component: string;
+    group?: DevfileCommandGroup;
+    label?: string;
+  };
+  composite?: {
+    commands: string[];
+    group?: DevfileCommandGroup;
+    label?: string;
+    parallel?: boolean;
+  };
+  exec?: {
+    commandLine?: string;
+    component: string;
+    env?: DevfileComponentEnv[];
+    group?: DevfileCommandGroup;
+    hotReloadCapable?: boolean;
+    label?: string;
+    workingDir?: string;
+  };
+  vscodeLaunch?: {
+    inline?: string;
+    uri?: string;
+    group?: DevfileCommandGroup;
+  };
+  vscodeTask?: {
+    inline?: string;
+    uri?: string;
+    group?: DevfileCommandGroup;
+  };
+  attributes?: { [attributeName: string]: string };
+  id: string
+}
+
+export interface DevfileComponentVolumeMount {
+  name?: string;
+  path?: string;
+}
+
+export interface DevfileComponentEnv {
+  name: string;
+  value: string;
+}
+
+export interface DevfileComponentEndpoint {
+  attributes?: { [attributeName: string]: string };
+  exposure?: 'public' | 'internal' | 'none';
+  name: string;
+  path?: string;
+  protocol?: 'http' | 'https' | 'ws' | 'wss' | 'tcp' | 'udp';
+  secure?: boolean;
+  targetPort: number;
+}
+
+export interface DevfileKubernetesComponent {
+  inlined?: string;
+  uri?: string;
+  endpoints?: DevfileComponentEndpoint[];
+}
+
+export interface DevfileKubernetesComponent {
+  inlined?: string;
+  uri?: string;
+  endpoints?: DevfileComponentEndpoint[];
+}
+
+export interface DevfileContainerComponent {
+  args?: string[];
+  command?: string[];
+  cpuLimit?: string;
+  cpuRequest?: string;
+  dedicatedPod?: boolean;
+  endpoints?: DevfileComponentEndpoint[];
+  env?: DevfileComponentEnv[];
+  image: string;
+  memoryLimit?: string;
+  memoryRequest?: string;
+  mountSources?: boolean;
+  sourceMapping?: string;
+  volumeMounts?: DevfileComponentVolumeMount[];
```

Comment:
well, these model interfaces should be dropped I believe since now we have up to date/correct model from @devfile/api

Reply:
good point! I'll fix that